### ISSUE TITLE
SqlDatabase.Inner.Tags isn't populated when loading databases from a SqlServer

### ISSUE THREAD

**Issue by rajbos** [Azure/azure-libraries-for-net#563]
SqlDatabase.Inner.Tags isn't populated when loading databases from a SqlServer.

``` C#
var sqlServers = await AzureManagementClient.SqlServers.ListAsync();
foreach (var sqlServer in sqlServers)
{
  ...
  var databases = await sqlServer.Databases.ListAsync();
  foreach (var database in databases)
  {
       var tags = database.Inner.Tags; // Tags == null
  }
}
```

Doublechecked with PowerShell that the resource does have tags on it via:
``` PowerShell
$resource = Get-AzureRmResource -Name $resourceName -ResourceGroupName $resourceGroupName
$resource.Tags
```

**Comment by hovsepm** [comment]
@rajbos this is a bug in the SQL Database API version 2014-04-01 that we rely on. The solution would be to migrate to the latest available Sql V3 however this is a very big functional and breaking change. In the meantime as a workaround, you can get tags by making Get call `sqlServer.Databases.GetById(database.Id)`. This will return you the full object with Tags. It will add one extra REST call to your communication flow though.

**Comment by hovsepm** [comment]
@rajbos BTW, you can also call `database.Refresh()` which will do the same `GetById` underneath and populate tags.

**Comment by jaredmoo** [comment]
> @rajbos this is a bug in the SQL Database API version 2014-04-01 that we rely on. The solution would be to migrate to the latest available Sql V3 however this is a very big functional and breaking change.

This is not correct. Looking at https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/specification/sql/resource-manager/readme.md you can see that package-composite-**v1** has 2015/servers which is the fixed version.

**Comment by archonandrewhunt** [comment]
What's the status of this ticket? It's 2021 and tags still do not get included in a call to `ListBySqlServerAsync`. Telling us to call refresh on each database returned is not viable considering we are listing thousands of databases. Can this get fixed properly?

**Comment by raschle** [comment]
Any updates on this?

**Comment by weidongxu-microsoft** [comment]
Database (not server) is still 2014
```
- Microsoft.Sql/stable/2014-04-01/databases.json
```

This SDK is in maintenance mode. New SDK is starting preview on https://aka.ms/azsdk/dotnet/mgmt (at present it does not have SQL yet).
